Contact form functionality

mrvanagas opened the pull request:
Added contact form. Form validation and error handling is handled by Formik:
![image](https://user-images.githubusercontent.com/71028004/169542457-811eb016-e3e2-4b7b-8d49-9a45be766390.png)
We can see, that if email in invalid, it will show an invalid email error, checking by Regex. We can also see, that comment input field is getting an error, since it is empty. Once all fields have some sort input, only then the form will allow to submit.

Once submitted, the inputs are stored in localStorage:
![image](https://user-images.githubusercontent.com/71028004/169542833-546ad131-a5bd-4115-a127-e9c3d7b537c2.png)

mrvanagas committed:
basic api functions

Added 3 basic API functions, also gets local coordinates

mrvanagas committed:
basic API functions

Basic API functions with 3 calls

mrvanagas committed:
Project restructure, added server

Restructured project structure to separate folders, added backend server to generate tokens.

mrvanagas committed:
removed token

Removed hardcoded token

mrvanagas committed:
Adjusted geolocator function

Geolocator now is called automatically in API.js file to get current current coordinates

Commit:
Auto weather info

Added logic to automatically show weather, based on current position, using the geolocator

Commit:
code modification

mrvanagas committed:
Removed hardcoded token, added other pages and router

Removed hard coded token, added several pages as placeholders, added router

mrvanagas committed:
Removed token

Removed hardcoded token

mrvanagas committed:
added 1st iteration of silent auth

Added first iteration of silent authentication, exploring other options with timed tokens

mrvanagas committed:
fixed files

fixed wrong file deletion

mrvanagas committed:
first iteration of search

First iteration of search functionality, first working with static data, in order to make the foundation of the search, will change to getting the info from API

mrvanagas committed:
adjustments

mrvanagas committed:
Added styled components

Refactored code to use styled components instead of regular css

mrvanagas committed:
added search from API

Added search from API, now need to figure out how to pass city id to the end point

mrvanagas committed:
removed unneeded file

mrvanagas committed:
Added city search

Added city search by entering part of the city name.

mrvanagas committed:
Merge branch 'Vanagas_Lukas' into city_search_function

mrvanagas committed:
Added dateHelper

Added datehelper, to convert date to weekdays, shortened styles.

mrvanagas committed:
Merge branch 'city_search_function' of https://github.com/mrvanagas/ReactTask into city_search_function

mrvanagas committed:
feat: added date formating

Added dateHelper file, to format the full date into weekday.

mrvanagas committed:
feat: added first iteration of debouncing

Added first iteration of debouncing the search in the city search page.

mrvanagas committed:
Feat: added contact page

Added contact page. Validation and form handling is done by Formik. When submitting fields, the info is set into localStorage

mrvanagas committed:
Added Comment title

mrvanagas committed:
added more conditions for input

JuliaShavyrova commented on client/src/Components/ContactForm.jsx:
```diff
@@ -0,0 +1,104 @@
+  if(!values.name) {
+      errors.name = 'Required'
+  } else if (values.name.length < 3) {
+    errors.name = 'Must be atleast 3 characters'
```
please fix typos in messages

JuliaShavyrova commented on client/src/Components/ContactForm.jsx:
```diff
@@ -0,0 +1,104 @@
+}
+
+const ContactForm = () => {
+  const formik = useFormik({
```
variable can be destructured

mrvanagas replied:
Destructured variable, also refactored the code, to make it more readable, by using `getFieldProps` method

mrvanagas committed:
Code refactor

Destructured variables, refactored some code, to make it shorter and more readable.

JuliaShavyrova commented:
contact form component looks good to me, but i cannot merge this PR, because it contains code from PR #209 that's not approved

mrvanagas committed:
Merge branch 'Vanagas_Lukas' into contact_form_functionality

mrvanagas commented:
Resolved conflicts, should be ready for merge

JuliaShavyrova commented on client/src/Components/ContactForm.jsx:
```diff
@@ -0,0 +1,104 @@
+};
+
+const ContactForm = () => {
+  const {handleSubmit, errors, touched, getFieldProps} = useFormik({
```
please don't forget to check formatting before commiting/creating PRs

mrvanagas replied:
Formatting should be done for this branch.

mrvanagas committed:
Ran formatter
